sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -223,6 +223,9 @@ abstract class Catalog {
    * If this table is cached as an InMemoryRelation, drop the original cached version and make the
    * new version cached lazily.
    *
+   * If the table's schema is inferred at runtime, infer the schema again and update the schema
```

Comment:
Yes unfortunately I find out about this one too late. I will add it to the release notes for 2.0 that this change will come.